### PR TITLE
[SERVER-1931] - Setting nomad port 4647 as default variable

### PR DESCRIPTION
**Pull request by circlery**
:gear: **Issue**
Sets nomad port 4647 as a default variable, as it's unlikely to change.

:white_check_mark: **Fix**
Creates a separate variable "server_port_nomad" in GCP and AWS modules, with a default of 4647, so the `:4647` does not need to be provided in the "server_endpoint" variable.

:question: **Tests**
In progress

- [ ] Passed _reality check_

### CODE REVIEW

**Comment by ryanwohara** [comment]
My only concern with this is the breaking change for existing customers. Not sure how to handle that gracefully. Maybe we could split the hostname by a colon and grab the first half?

File: nomad-aws/variables.tf [hunk 1]
```diff
@@ -21,7 +21,13 @@ variable "security_group_id" {
 
 variable "server_endpoint" {
   type        = string
-  description = "Domain and port of RPC service of Nomad control plane (e.g example.com:4647)"
+  description = "Domain of RPC service of Nomad control plane (e.g example.com)"
+}
+
+variable "server_port_nomad" {
+  type        = number
+  description = "Port that the server endpoint listens on for nomad connections (defaults to 4647)."
```
Comment (lloydChris) [inline nomad-aws/variables.tf:29]: nit: Probably dont need to write `defaults to 4646` here or in GCP

**Comment by lloydChris** [comment]
> My only concern with this is the breaking change for existing customers. Not sure how to handle that gracefully. Maybe we could split the hostname by a colon and grab the first half?

The breaking change is notable.  I could get behind a major version bump to account for it.

File: nomad-aws/examples/basic/main.tf [hunk 2]
```diff
@@ -38,7 +38,7 @@ module "nomad-aws" {
 
   # Location of your Nomad server endpoint. This should be exposed from your
   # server installation via a load balancer service.
-  server_endpoint = "example.com:4647"
+  nomad_server_hostname = "example.com"
```
Comment (atulsingh0) [inline nomad-aws/examples/basic/main.tf:41]: the name `nomad_server_hostname` seems to suggest, this is different than the application DNS name. We should use - `server_hostname`

File: nomad-aws/variables.tf (hunk 1 above)
Comment (atulsingh0) [reply nomad-aws/variables.tf]: I don't think, we need this variable at all.

**Review by lloydChris — APPROVED** [review]

**Review by ryanwohara — APPROVED** [review]

**Pull request merged**